def personal_sum(numbers):
    result = 0
    incorrect_data = 0
    for item in numbers:
        try:
            # Попробуем прибавить число к результату
            result += item
        except TypeError:
            # Если возникает TypeError, увеличиваем счётчик incorrect_data и выводим сообщение
            print(f'Некорректный тип данных для подсчёта суммы - {item}')
            incorrect_data += 1
    return result, incorrect_data

def calculate_average(numbers):
    try:
        # Получаем сумму чисел и количество некорректных данных
        result, incorrect_data = personal_sum(numbers)
        # Вычисляем среднее арифметическое, обработав ZeroDivisionError
        return result / (len(numbers) - incorrect_data)
    except ZeroDivisionError:
        # Возвращаем 0, если деление на 0
        return 0
    except TypeError:
        # Выводим сообщение и возвращаем None, если передан некорректный тип данных
        print('В numbers записан некорректный тип данных')
        return None

# Тестирование функций
print(f'Результат 1: {calculate_average("1, 2, 3")}')
print(f'Результат 2: {calculate_average([1, "Строка", 3, "Ещё Строка"])}')
print(f'Результат 3: {calculate_average(567)}')
print(f'Результат 4: {calculate_average([42, 15, 36, 13])}')
